import-export-cli/integration/testutils/environment_testUtils.go

```diff
@@ -411,3 +411,234 @@ func validateMutualSSLCerts(t *testing.T, apiParams *Params, path string) {
+	if strings.EqualFold(endPointTypeInParams, "rest") {
+		endPointTypeInParams = "http"
+	} else if strings.EqualFold(endPointTypeInParams, "soap") {
+		endPointTypeInParams = "address"
+	}
+	assert.Equal(t, endPointTypeInParams, api.GetEndpointType())
+}
+
+func validateEndpointUrl(t *testing.T, apiParams *Params, api *apim.API, endpointType string) {
+	var endpointTypeRecord map[string]interface{}
+
+	if strings.EqualFold(endpointType, "production") {
+		endpointTypeRecord = apiParams.Environments[0].Configs.Endpoints.Production
+		assert.Equal(t, endpointTypeRecord["url"], api.GetSandboxURL())
+	}
+	if strings.EqualFold(endpointType, "sandbox") {
+		endpointTypeRecord = apiParams.Environments[0].Configs.Endpoints.Sandbox
+		assert.Equal(t, endpointTypeRecord["url"], api.GetSandboxURL())
+	}
+}
+
+func ValidateHttpEndpointWithoutLoadBalancingAndFailover(t *testing.T, api *apim.API, apiParams *Params, importedAPI *apim.API) {
+	t.Helper()
+	//Validate EndPoint Type
+	validateEndpointType(t, apiParams, importedAPI)
+
+	// Validate endpoint type URLs
+	validateEndpointUrl(t, apiParams, importedAPI, "production")
+	validateEndpointUrl(t, apiParams, importedAPI, "sandbox")
+
+	same := "override_with_same_value"
+	api.SetEndpointType(same)
+	api.SetProductionURL(same)
+	api.SetSandboxURL(same)
+	importedAPI.SetEndpointType(same)
+	importedAPI.SetProductionURL(same)
+	importedAPI.SetSandboxURL(same)
+
+	ValidateAPIsEqual(t, api, importedAPI)
+}
+
+func ValidateHttpEndpointWithLoadBalancing(t *testing.T, api *apim.API, apiParams *Params, importedAPI *apim.API) {
+	t.Helper()
+	//Validate EndPoint Type
+	assert.Equal(t, "load_balance", importedAPI.GetEndpointType())
+
+	endPointConfigInApi := importedAPI.EndpointConfig
+
+	// Validate Algorithm class
+	algoClassNameInParams := apiParams.Environments[0].Configs.LoadBalanceEndpoints.AlgorithmClassName
+	algoClassNameInApi := endPointConfigInApi.(map[string]interface{})["algoClassName"].(string)
+	assert.Equal(t, algoClassNameInParams, algoClassNameInApi)
+
+	// Validate Session TimeOut
+	sessionTimeOutInParams := apiParams.Environments[0].Configs.LoadBalanceEndpoints.SessionTimeout
+	sessionTimeOutInApi := endPointConfigInApi.(map[string]interface{})["sessionTimeOut"].(string)
+	assert.Equal(t, strconv.Itoa(sessionTimeOutInParams), sessionTimeOutInApi)
+
+	// Validate Production endpoints
+	productionEndpointsInParams := apiParams.Environments[0].Configs.LoadBalanceEndpoints.Production
+	productionEndpointsInApi := endPointConfigInApi.(map[string]interface{})["production_endpoints"].([]interface{})
+	if strings.EqualFold(apiParams.Environments[0].Configs.LoadBalanceEndpoints.SessionManagement, "transport") {
+		for i := 0; i < len(productionEndpointsInParams); i++ {
+			singleProductionEndpointInApi := productionEndpointsInApi[i].(map[string]interface{})
+			assert.Equal(t, productionEndpointsInParams[i]["url"], singleProductionEndpointInApi["url"])
+		}
+	}
```

Comment:
Shall we move this repetitive code segment to another function? (this is repeated in 496 to 501)